servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp

```diff
@@ -608,6 +608,9 @@ void SceneShaderForwardClustered::init(RendererStorageRD *p_storage, const Strin
 		//builtins
 
 		actions.renames["TIME"] = "scene_data.time";
+		actions.renames["PI"] = "3.1415926535";
+		actions.renames["TAU"] = "6.2831853072";
+		actions.renames["E"] = "2.7182818285";
```

Comment:
We already have those defined for the whole engine in `core/math/math_defs.h`, so it would be best to reuse those definitions instead of re-hardcoding those constants with a different precision.

You can convert to string with `_MKSTR` (defined in `core/typedefs.h`, should already be included everywhere so you don't need to include it explicitly):

```cpp
#include "core/math/math_defs.h"

...

actions.renames["PI"] = _MKSTR(Math_PI);
```